google/cloud/spanner_dbapi/client_side_statement_executor.py

```diff
@@ -33,8 +47,43 @@ def execute(connection: "Connection", parsed_statement: ParsedStatement):
     :param parsed_statement: parsed_statement based on the sql query
     """
     if parsed_statement.client_side_statement_type == ClientSideStatementType.COMMIT:
-        return connection.commit()
+        connection.commit()
+        return None
     if parsed_statement.client_side_statement_type == ClientSideStatementType.BEGIN:
-        return connection.begin()
+        connection.begin()
+        return None
     if parsed_statement.client_side_statement_type == ClientSideStatementType.ROLLBACK:
-        return connection.rollback()
+        connection.rollback()
+        return None
+    if (
+        parsed_statement.client_side_statement_type
```

Comment:
Related, very minor thing:

In Python, everything is interpreted and there's no JIT compiler doing type inference, so every time you do `a.b`, the Python interpreter has to actually interpret that logic.  Specifically -- go look up the type of `a`, look up and run code associated with any overrides on `a`'s `.` operator (for example properties are implemented under the hood as an overload to the dot operator that checks whether the name of the field that you're looking up is the same as the name of a property on the class and, if so, replaces that lookup with a call to the underlying getter(/setter(/deleter)) function), then if still applicable go look up `b` in `a`'s member dictionary.

As a result:

```
a.b
a.b
a.b
a.b
(... lots of times ...)
```

is actually slightly-but-measurably slower than

```
c = a.b
c
c
c
c
(... lots of times ...)
```

In most other languages, unless a pointer dereference is required (which is still cheap), nested member-variable access is zero cost at runtime; all of this resolution is sorted out at compile time or by a JIT compiler.

The performance difference is not _usually_ enough (and Python code is usually not performance-sensitive enough) that it matters.  But it gently nudges Python to tend to use simple variables rather than complex nested structures.

Reply:
Defined a variable `statement_type`. Thanks for the explanation, was not aware of it. Will try to remember and take care of this in future code